test-ciphers.sh

```diff
@@ -1,56 +1,70 @@
-#!/usr/bin/env bash
+#!/bin/bash
 
-CONSOLE_URL=$(oc get console.config.openshift.io cluster --template '{{.status.consoleURL}}')
-if [ -z "${CONSOLE_URL}" ]
-then
-  echo "no console_url, are you connected to a cluster?"
-  exit 1
+set -o errexit
+set -o nounset
+set -o pipefail
+
+oc project openshift-console
+
+CONSOLEPOD=$(oc get pods --template '{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}' | grep --max-count 1 console)
+CONSOLE_POD_IP=$(oc get pod "${CONSOLEPOD}" --template '{{.status.podIP}}')
+
+echo "pod ${CONSOLEPOD} at ${CONSOLE_POD_IP}"
+
+POD_NAME="test-console-ciphers"
+CONFIGMAP_NAME="${POD_NAME}"
+FILE_NAME="test-ciphers-from-pod.sh"
+FILE_PATH="/scripts"
+
+
+# clean up after last run
+oc delete pod "${POD_NAME}"
+oc delete configmap "${CONFIGMAP_NAME}"
+
+# put the test script in a configmap in the console namespace
+oc create configmap "${CONFIGMAP_NAME}" --from-file="./test/${FILE_NAME}"
+
+# creates a POD to run the test script in the console namespace
+# passes $CONSOLE_POD_IP as $SERVER environment variable
+cat <<EOF | oc create -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  namespace: openshift-console
```

Comment:
Does this pod need to live in the openshift-console namespace? I feel like it would be better to create and delete a test namespace, then there's no clean up to worry about.

Reply:
works for me.